python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_ledger_orchestrator/_ledger_orchestrator_manager.py

```diff
@@ -0,0 +1,333 @@
+
+    def _get_task_ledger_full_prompt(self, task: str, team: str, facts: str, plan: str) -> str:
+        return ORCHESTRATOR_TASK_LEDGER_FULL_PROMPT.format(task=task, team=team, facts=facts, plan=plan)
+
+    def _get_progress_ledger_prompt(self, task: str, team: str, names: List[str]) -> str:
+        return ORCHESTRATOR_PROGRESS_LEDGER_PROMPT.format(task=task, team=team, names=", ".join(names))
+
+    def _get_task_ledger_facts_update_prompt(self, task: str, facts: str) -> str:
+        return ORCHESTRATOR_TASK_LEDGER_FACTS_UPDATE_PROMPT.format(task=task, facts=facts)
+
+    def _get_task_ledger_plan_update_prompt(self, team: str) -> str:
+        return ORCHESTRATOR_TASK_LEDGER_PLAN_UPDATE_PROMPT.format(team=team)
+
+    def _get_final_answer_prompt(self, task: str) -> str:
+        return ORCHESTRATOR_FINAL_ANSWER_PROMPT.format(task=task)
+
+    @event
+    async def handle_start(self, message: GroupChatStart, ctx: MessageContext) -> None:
+        """Handle the start of a group chat by selecting a speaker to start the conversation."""
+        assert message is not None and message.message is not None
+
+        # Log the start message.
+        await self.publish_message(message, topic_id=DefaultTopicId(type=self._output_topic_type))
+
+        # Create the initial task ledger
+        #################################
+        self._task = self._content_to_str(message.message.content)
+        planning_conversation: List[LLMMessage] = []
+
+        # 1. GATHER FACTS
+        # create a closed book task and generate a response and update the chat history
+        planning_conversation.append(
+            UserMessage(content=self._get_task_ledger_facts_prompt(self._task), source=self._name)
+        )
+        response = await self._model_client.create(planning_conversation)
+
+        assert isinstance(response.content, str)
+        self._facts = response.content
+        planning_conversation.append(AssistantMessage(content=self._facts, source=self._name))
+
+        # 2. CREATE A PLAN
+        ## plan based on available information
+        planning_conversation.append(
+            UserMessage(content=self._get_task_ledger_plan_prompt(self._team_description), source=self._name)
+        )
+        response = await self._model_client.create(planning_conversation)
+
+        assert isinstance(response.content, str)
+        self._plan = response.content
+
+        # Kick things off
+        self._n_stalls = 0
+        await self._reenter_inner_loop()
+
+    @event
+    async def handle_agent_response(self, message: GroupChatAgentResponse, ctx: MessageContext) -> None:
+        self._message_thread.append(message.agent_response.chat_message)
+        await self._orchestrate_step()
+
+    @event
+    async def handle_reset(self, message: GroupChatReset, ctx: MessageContext) -> None:
+        # Reset the group chat manager.
+        await self.reset()
+
+    async def select_speaker(self, thread: List[AgentMessage]) -> str:
```

Comment:
is this function needed?

Reply:
Nope. Unless we subclass the BaseGroupChatManager